### PR TITLE
Logger: Merge ESLoggerFactory into Loggers

### LINKED ISSUE #32174: Logging: Clean up logger building
We build loggers in about 15 different ways right now. After #31588 we'll only really need about 3 different ways. We should remove all of the confusing "variety".

### PR DESCRIPTION
**Pull request by nik9000**
`ESLoggerFactory` is now not particularly interesting and simple enought
to fold entirely into `Loggers. So let's do that.

Closes #32174

### CODE REVIEW

**Review by cbuescher — COMMENTED** [review]
@nik9000, great change, I left just one question regarding the prefix null-checks and where/when this should be added. I think I'm happy either way, just wanted to ask.

**Comment by cbuescher** [comment]
Forgot one question about the intended 6.6 backport: as this is a breaking Java change, I was wondering if folks writing plugins etc. might currently use ESLoggerFactory in its current form and if we should deprecate on 6.6. instead of immediate removal. If that was the intention anyway then nevermind.

File: server/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-         * because we expect to eventually migrate away from needing this
-         * method entirely.
-         */
-        return getLogger(prefix, LogManager.getLogger(clazz.getName()));
-    }
-
-    static Logger getLogger(String prefix, Logger logger) {
-        /*
-         * In a followup we'll throw an exception if prefix is null or empty
-         * redirecting folks to LogManager.getLogger.
```
[the remark this answers is not in the document]
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/common/logging/ESLoggerFactory.java]: I hadn't opened a task for it. To be honest I'd forgotten about it and saw it when doing this. I'm happy to switch the `assert` to a hard check.

**Comment by nik9000** [comment]
About this being breaking for plugins: I'm lumping this in with the "our plugin API isn't stable" business. It is kind of a step to getting it stable because we no longer point folks to our own logging abstraction. But that is a fairly weak excuse to be honest. I think this is just what writing a plugin is like for now. One day, we'll have a stable plugin API. But we can't not change all this internal stuff just because plugins might touch it.

**Review by cbuescher — APPROVED** [review]
@nik9000 thanks for the changes, LGTM

**Comment by nik9000** [comment]
Thanks for reviewing @cbuescher! I finally finished the backport.

**Pull request merged**